rocky/tools/models.py

```diff
@@ -104,8 +132,20 @@ def delete(self, *args, **kwargs):
 
         super().delete(*args, **kwargs)
 
+    def clean(self):
+        if self.code in DENY_ORGANIZATION_CODES:
+            raise ValidationError(
```

Comment:
This validation error should be handled in organization_add.py. The ServiceException is handled there as well and perhaps adding the ValidationError to the except is enough, but now a user without DEBUG=True will have no idea why they cannot create their organization.

Reply:
@Donnype I put this directly in the model because we also have an /api connection to create organizations, also in the django admin. Then I would need to create separate validations for each view.

Reply:
The form used in `organization_add` is different as well as those used for the api and admin.

Reply:
Throwing the error here is fine! But organization_add.py will now give us an error page as your screenshot shows. When debug is turned off, this will be a generic 500 error without the user knowing that he used a denied name. So I would add an 'except ValidationError' in that view and tell the user either the denylist or at least that they added a denied name either through messaging or an error page.

(How the API handles it is a different scenario indeed, which perhaps should be checked as well but might be less important because of the limited usage (in terms of users) of the API and the fact that the current API users probably have access to logs as well.)

Reply:
The 500 error is at the API url not from `organization_add` view.

Reply:
My bad